### DISCUSSION TITLE
Create pull request and add labels in a single request

### DISCUSSION THREAD

**Discussion by sorenlouv** [community/community#24724] [Apps, API and Webhooks]
<p>Hi there,</p>
<p>I’m trying to create a pull request with labels in a single request via Github’s REST (v3) API.<br>
From <a href="https://developer.github.com/v3/pulls/#create-a-pull-request" rel="nofollow noopener">the docs</a> it seems like it is not possible to add labels while creating a pull request.<br>
This surprises me, since it’s possible to <a href="https://developer.github.com/v3/issues/#create-an-issue" rel="nofollow noopener">create an issue and add labels with a single request</a>.</p>
<p>To be sure I did try creating a PR with <code>labels</code> in the body but it did not work (PR created but no label added):</p>
<pre><code class="lang-auto">curl --request POST \
  --url https://api.github.com/repos/sqren/backport-demo/pulls \
  --data '{
	"title": "My PR title",
	"body": "Some description...",
	"head": "sqren:7.x",
	"base": "master",
	"labels": ["my-label"]
}'
</code></pre>
<p>Is this possible with the current REST (v3) api or via the graphql (v4) api?</p>
<p>Thanks!</p>

**Comment by lee-dohm** [comment]
<p>You are correct that there isn’t a way to do what you’re describing using the REST API.</p>
<p>Additionally, there isn’t a way to add labels via <a href="https://developer.github.com/v4/mutation/createpullrequest/">the GraphQL API <code>createPullRequest</code> mutation</a>. You can submit multiple mutations in one request via GraphQL, but I have not been able to discover a way to use the output of one mutation as an input of a successive mutation in the same request via GraphQL. Because of this, it would appear that you would have to use two separate requests even via the GraphQL API.</p>
<p>Perhaps someone more experienced with GraphQL could comment on a way to make it happen.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/sqren">@sqren</a> Thanks for being here, while I have no input as to how to add this programmatically at creation via the API (I’m not sure this is possible. When using the API, adding labels takes two steps: <em>PR saved first then labels are saved to the database vs in the UI is done in one step</em>)<br>
In anycase, I’d like to share that the <a href="https://cli.github.com/">GitHub CLI</a> has included the ability to add metadata such as labels at PR creation in one of its <a href="https://github.com/cli/cli/releases/tag/v0.8.0">latest releases</a> maybe check it out.</p>
<p>I hope that helps!</p>

**Reply by shinebayar-g** [reply]
This doesn't seem to work. Using Github CLI allows to define labels while creating a Pull Request. However on the github workflow side, `github.event.pull_request.labels` stills comes empty.

**Comment by sorenlouv** [comment, marked as answer]
<p>Thanks both. I’ll have to make due with multiple requests (In addition to adding labels I also need to add assignees to the PR).</p>
<p>Would be great if this could be done at creation time like it’s possible with issues.</p>

**Comment by GuillaumeFalourd** [comment]
<p>Just completing the answer as I just experienced something similar:</p>
<p>To add labels to a PR with the Github API, it’s necessary to call 2 endpoints:</p>
<ul>
<li>One for creating the Pull Request:</li>
</ul>
<pre><code class="lang-auto">curl --request POST \
  --url https://api.github.com/repos/{repo_owner}/{repo_name}/pulls \
  --data '{
	"title": "My PR title",
	"body": "Some description...",
	"head": "{head}",
	"base": "master"
}'

</code></pre>
<ul>
<li>One for adding the Labels (example with the <code>enhancement</code> label):</li>
</ul>
<pre><code class="lang-auto">curl --location --request POST 'https://api.github.com/repos/{repo_owner}/{repo_name}/issues/{pr_number}/labels' \
--header 'Accept: application/vnd.github.v3+json' \
--header 'Authorization: Bearer {GITHUB_PAT}' \
--header 'Content-Type: application/json' \
--data-raw '{
    "labels": ["enhancement"]
}'
</code></pre>
<p>The <code>labels</code> field represents the names of the labels to add to the issue / PR.</p>
<p><em>Note 1: You can also pass an empty array to remove all labels.</em></p>
<p><em>Note 2: You can also pass a single label as a string or an array of labels directly, but GitHub recommends passing an object with the labels key.</em></p>

**Comment by DRMADOOK2020** [comment]
Hello my friend,  can you say me about my web or phone problem?
